src/Pages/SettingPage/SettingPage.tsx

```diff
@@ -1,121 +1,146 @@
 import LanguageButton from "../../Components/LanguageButton/LanguageButton";
-import { useAlert } from "../../Utils/AlertContext.tsx";
+import { isAuthenticated, logout } from "../../Utils/Auth/AuthUtil";
+import { useAlert } from "../../Utils/AlertContext";
 import "./SettingPage.css";
 
-function SettingPage() {
+function SettingsPage() {
   const { t } = useTranslation();
   const { showAlert } = useAlert();
   const [uname_validated, setUnameValidated] = useState(false);
   const [uname, setUname] = useState("");
   const [password, setPassword] = useState("");
-  const login = () => {
+  const [auth, setAuth] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setAuth(isAuthenticated());
+  }, []);
+
+  const login = async () => {
+    setIsLoading(true);
     const form = new FormData();
     form.append("username", uname);
     form.append("password", password);
-    if (process.env.VITE_APP_ACTIVATE_USING_JSON == "true") {
+    if (process.env.VITE_APP_ACTIVATE_USING_JSON === "true") {
       document.cookie = `auth=${btoa(uname + ":" + password)}`;
       showAlert(t("loggedIn"), "confirm");
+      setIsLoading(false);
+      window.location.reload();
       return;
     }
-    fetch(process.env.VITE_API_URL + "/login", {
-      method: "POST",
-      body: form,
-      headers: {
-        Authorization: "Basic " + btoa(uname + ":" + password),
-      },
-    })
-      .then((r) => {
-        if (r.status !== 200) {
-          fetch(process.env.VITE_API_URL + "/signup", {
+    try {
+      const loginResponse = await fetch(process.env.VITE_API_URL + "/login", {
+        method: "POST",
+        body: form,
+        headers: {
+          Authorization: "Basic " + btoa(uname + ":" + password),
+        },
+      });
+      if (loginResponse.status !== 200) {
+        const signupResponse = await fetch(
+          process.env.VITE_API_URL + "/signup",
+          {
             method: "POST",
             body: form,
             headers: {
               Authorization: "Basic " + btoa(uname + ":" + password),
             },
-          })
-            .then((r) => {
-              if (r.status !== 201) {
-                r.json().then((data) => {
-                  showAlert(data.error, "error");
-                });
-              } else {
-                document.cookie = `auth=${btoa(uname + ":" + password)}`;
-                showAlert(t("loggedIn"), "confirm");
-              }
-            })
-            .catch((e) => {
-              showAlert(e, "error");
-            });
+          },
+        );
+        if (signupResponse.status == 401) {
+          const data = await signupResponse.json();
+          showAlert(data.error, "error");
         } else {
           document.cookie = `auth=${btoa(uname + ":" + password)}`;
-          showAlert(t("registered"), "confirm");
+          showAlert(t("loggedIn"), "confirm");
+          setAuth(true);
```

Comment:
I suggest setAuth(isAuthenticated()) for good measure.

Reply:
```typescript
  const finalizeAuth = (messageKey: string) => {
    authLogin(uname, password);
    showAlert(t(messageKey), "confirm");
    setAuth(true);
    window.location.reload();
  };
```

Here we should set setAuth(isAuthenticated()). But this is okay. Resolving this.